### PR TITLE
feat: migrate the app to movie db api

### PR DESCRIPTION
**Pull request by cindy-sf**

### CODE REVIEW

File: src/components/MovieCard/index.tsx [hunk 1]
```diff
@@ -22,21 +24,24 @@ const ImageWrapper = styled.View`
 `
 
 interface Props {
-  navigation: NavigationContainerRef
-  movie: Movie | SearchData
+  navigation: any
```
Comment (j0nathansf) [inline src/components/MovieCard/index.tsx]: You failed to type navigation ?
Why NavigationContainerRef no longer works ?

File: src/views/Movies/index.tsx [hunk 2]
```diff
@@ -40,86 +35,92 @@ interface APIMovieList {
...
+  } as const
 
   useEffect(() => {
-    const fetchData = async (): Promise<void> => {
+    const fetchData = async () => {
       try {
-        setIsDataFetching(true)
-
-        // Get movies list and all genders
+        // fetch popular and upcoming movies
```
Comment (fr-jsf) [inline src/views/Movies/index.tsx:67]: remove comments in this file if everything works

File: src/views/Search/index.tsx [hunk 3]
```diff
@@ -41,18 +44,18 @@ const Search = ({ navigation, route }: Props) => {
...
     }
-  }, [])
+  }, [route.params.search])
 
   const fetchMoviesBySearch = async (search: string): Promise<void> => {
     try {
       setIsDataFetching(true)
       const searchResponse = await fetch(
-        `https://api.betaseries.com/search/movies?key=${API_KEY}&text=${search}&limit=10`
+        `https://api.themoviedb.org/3/search/movie?api_key=${MOVIE_DB_API_KEY}&language=fr-FR&query=${search}&page=1`
```
Comment (j0nathansf) [inline src/views/Search/index.tsx:53]: why not use movieAPIUrls here  ?
src/constants/index.ts

**Review by j0nathansf — DISMISSED** [review]
I made some comments but overall it's very good

File: src/components/MovieCard/index.tsx (hunk 1 above)
Comment (cindy-sf) [reply src/components/MovieCard/index.tsx]: Oh yes, I forgot.
I have to type correctly the navigation with "push" method 👍

File: src/views/Search/index.tsx (hunk 3 above)
Comment (cindy-sf) [reply src/views/Search/index.tsx:53]: Good point

**Review by j0nathansf — APPROVED** [review]

**Pull request merged**